app/services/assignment_service.py

```diff
@@ -331,51 +371,64 @@ def get_adjusted_due_date(self) -> datetime | None:
             deferred_time = timedelta(0)
             extra_time = timedelta(0)
 
-        return self.assignment.due_date + deferred_time + extra_time + self.student.base_extra_time
-
-    def get_is_available(self) -> bool:
-        if not self.assignment.is_created: return False
+        return assignment_due_date + deferred_time + extra_time + self.student_model.base_extra_time
 
+    def _get_is_available(self) -> bool:
+        adjusted_available_date = self.get_adjusted_available_date()
+        if adjusted_available_date is None: return True
+        current_timestamp = self.session.scalar(func.current_timestamp())
+        return current_timestamp >= adjusted_available_date
```

Comment:
this line will error if `self.get_adjusted_available_date()` returns None. which is possible in the current implementation

Reply:
there's a null check (378)? not sure what you mean

Reply:
literally didnt see that if. ignore me I need my glasses checked